### PR TITLE
Invert code and style analysis configuration

### PR DESCRIPTION
**Pull request by richlander**
I've been playing with Jellyfin with .NET 6 for performance testing. I'd also like to look at improving Jellyfin performance by adopting various .NET 6 SDK feature (like ready to run compilation). I'd also like to help by fixing a bunch of your SA/CA and nullable warnings. Before doing that, it makes sense to invert the way that those are configured, to opt-out instead of opt-in. That's what this change does.

The projects where I added `<TreatWarningsAsErrors>false</TreatWarningsAsErrors>` are the ones generating all the warnings. It makes it easy to see where the technical debt is. The goal would be to fix all the warnings for each project and then remove that opt-out property.

I disabled analysis for DvdLib and RSSDP projects per advice in Discord. In short, the model would be this:

- `<TreatWarningsAsErrors>false</TreatWarningsAsErrors>` for projects where analysis clean is desired.
- `<Nullable>disable</Nullable>` and `<AnalysisMode>AllDisabledByDefault</AnalysisMode>` where there is no plan to get analysis clean.

If you folks are willing to take this PR, I'm happy to start working on fixing the warnings. I've done this before so know what is involved in fixing hundreds of warnings.

Complementary to https://github.com/jellyfin/jellyfin/issues/2149.

FYI: I am a member of the .NET Team.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<Project>
+
+  <PropertyGroup>
+    <Nullable>enable</Nullable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)/jellyfin.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+
+
```
Comment (Bond-009) [inline Directory.Build.props:9]:
```suggestion
```

File: DvdLib/DvdLib.csproj [hunk 2]
```diff
@@ -13,7 +13,8 @@
     <TargetFramework>net5.0</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AnalysisMode>AllDisabledByDefault</AnalysisMode>
+    <Nullable>Disable</Nullable>
```
Comment (Bond-009) [inline DvdLib/DvdLib.csproj:17]: `enable` is lowercase in the Directory.Build.props, maybe this should be too to be consistent?
```suggestion
    <Nullable>disable</Nullable>
```

File: Emby.Dlna/Emby.Dlna.csproj [hunk 3]
```diff
@@ -20,8 +20,7 @@
     <TargetFramework>net5.0</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <Nullable>enable</Nullable>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (Bond-009) [inline Emby.Dlna/Emby.Dlna.csproj:23]: Leave this true for Release builds (same for all other projects)

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Dlna/Emby.Dlna.csproj (hunk 3 above)
Comment (richlander) [reply Emby.Dlna/Emby.Dlna.csproj]: Yup. You are right. I didn't quite grasp what as going on with those style analyzers at first glance.

File: RSSDP/RSSDP.csproj [hunk 4]
```diff
@@ -13,7 +13,9 @@
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AnalysisMode>AllDisabledByDefault</AnalysisMode>
+    <Nullable>Disable</Nullable>
```
Comment (Bond-009) [inline RSSDP/RSSDP.csproj:17]:
```suggestion
    <Nullable>disable</Nullable>
```

**Review by Bond-009 — APPROVED** [review]
Not really a fan of the 2x `TreatWarningsAsErrors` in some projects, but good enough

**Review by barronpm — APPROVED** [review]

**Pull request merged**